one_fm/api/tasks.py

```diff
@@ -1684,6 +1684,10 @@ def mark_daily_attendance(start_date, end_date):
 					})
 			except Exception as e:
 				errors.append(str(frappe.get_traceback()))
+    
+		#Approve all open timesheets whose start date and end date is today
+		frappe.db.sql(f"""UPDATE `tabTimesheet` SET workflow_state = 'Approved' where start_date = '{getdate()}' and end_date = '{getdate()}' and workflow_state = 'Open' """)

```

Comment:
This is not the solution, it will not create attendance based on timesheet nor approve attendance for the set date.